# **ATAC-seq Project:**

## Raw sequence data processing. 

The main script is bamprocess.sh. It generate bed files and corrected bam files from fastq 

## Peak calling:

The main script is peakcalling.sh. It calls for regions which have the highest read density compared to background reads density. Within these regions, the highest read density locus is the peak summit.

## Peak merging and averaging:

The main script is averagepeak.sh. This script takes in all peak files called for each tissues (4 tissues total). Then, it process the script accodring to the averaging method in the paper. 

## Euchromatin peak filter and peak annotation: 

The main script is peakannotation.sh. It is used to annotated the peaks that are filtered using the euchromatin boundaries described in the paper. 

## Peak quality control plot:

The main script is peakqualityplot.r. The script will output all of the peak quality control plots in the paper. The inputs for the script are generated by peakqualitydatageneration.sh

## Normalization for differences between genotypes and tissues for all samples:

The main scripts are fragfile1.sh, fragfile2.sh, and fragfile3.sh . These are used to calculate for weights for each fragment lengths after removal of SV spanning fragments. If weights are calculated for all fragments, the bedtools intersect command in fragfile1.sh need to be removed. This command is used to remove all SV spanning fragments. 

## C and L matrix calculation:

The main scripts are CLmatrix.sh and CL.py. The CLmatrix.sh is the script that process fragment files and calculate for C and L by locus using the CL.py script. The model for C and L are:

a/ Ci = sum(Wi)
b/ Li = sum(Wi * FLi )/ sum(Wi)

Ci stands for C value at position i. Wi stands for weight at position i. FLi is the fragment lenght at position i and its weight Wi. 

## C and L matrices statistical test:

The main scripts are CLanova1.sh, CLanova2.sh, CLanova.r



